test/LibraryManager.IntegrationTest/Services/InstallDialogTestExtension.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using Microsoft.Test.Apex.Services;
+using Microsoft.Test.Apex.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.Web.LibraryManager.Vsix.UI;
+
+namespace Microsoft.Web.LibraryManager.IntegrationTest.Services
+{
+    public class InstallDialogTestExtension : VisualStudioInProcessTestExtension<object, InstallDialogVerifier>
+    {
+        /// <summary>
+        /// Add client side libraries dialog test extension for interaction with visual studio inprocess types
+        /// </summary>
+        public IInstallDialogTestContract InstallDialog
+        {
+            get
+            {
+                return this.ObjectUnderTest as IInstallDialogTestContract;
+            }
+        }
+
+        public void SetLibrary(string library)
+        {
+            UIInvoke(() =>
+            {
+                this.InstallDialog.Library = library;
+            });
+        }
+
+        public void WaitForFileSelections()
+        {
+            WaitFor.IsTrue(() =>
+            {
+                return UIInvoke(() =>
+                {
```

Comment:
Would it make more sense for the UIInvoke to be outwise the WaitFor.IsTrue to only do one thread switch?

Reply:
As discussed offline, switching it causes the UI to hang as messages don't get pumped. I tried to switch to JTF, but it fails in restore step.

Keeping this as it.

---
In reply to: [205452506](https://github.com/aspnet/LibraryManager/pull/332#discussion_r205452506) [](ancestors = 205452506)